keps/sig-cli/20201022-kubectl-xdg-base-dir.md

```diff
@@ -0,0 +1,193 @@
+| Need to update | cmd/kubeadm/app/cmd/reset.go                                                            | contains reference to $HOME/.kube       |
+| Need to update | cmd/kubeadm/app/cmd/completion.go                                                       | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go                    | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/cli-runtime/pkg/genericclioptions/client_config.go                   | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/sample-controller/README.md                                          | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/kubectl/pkg/cmd/config/create_cluster.go                             | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/kubectl/pkg/cmd/config/create_authinfo.go                            | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/kubectl/pkg/cmd/testing/fake.go                                      | contains reference to $HOME/.kube/cache |
+| Need to update | staging/src/k8s.io/kubectl/pkg/cmd/completion/completion.go                             | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/kubectl/pkg/generated/bindata.go                                     | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/sample-apiserver/README.md                                           | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/client-go/util/homedir/homedir.go                                    | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery.go                  | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/client-go/examples/dynamic-create-update-delete-deployment/main.go   | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/client-go/examples/dynamic-create-update-delete-deployment/README.md | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/client-go/examples/create-update-delete-deployment/main.go           | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/client-go/examples/create-update-delete-deployment/README.md         | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/client-go/examples/out-of-cluster-client-configuration/main.go       | contains reference to $HOME/.kube       |
+| Need to update | staging/src/k8s.io/client-go/tools/clientcmd/loader.go                                  | contains reference to $HOME/.kube       |
```

Comment:
I think it's worth pointing out that code like this: https://github.com/kubernetes-sigs/controller-runtime/blob/cb7f85860a8cde7259b35bb84af1fdcb02c098f2/pkg/client/config/config.go#L129  exists in the wild, making using of expected paths directly (since they're exported).  You're gonna need to be real careful in how you change that, since it's real easy to make something fail even though there's not change at compile time w/o it being obvious why.